### PR TITLE
Make Language<>Extension mappings configurable

### PR DESCRIPTION
**Pull request by julianthome**
This MR adds an interface to make the language<>extensions mappings configurable.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 	<groupId>com.github.magpiebridge</groupId>
 	<artifactId>magpiebridge</artifactId>
-	<version>0.1.4-SNAPSHOT</version>
+	<version>0.2.0-SNAPSHOT</version>
```
Comment (linghuiluo) [inline pom.xml]: Why are you suggesting 0.2.0-SNAPSHOT? The next version would be 0.1.5.  Can you change it to 0.1.5-SNAPSHOT? I will make a release to maven central.

File: src/main/java/magpiebridge/core/DefaultLanguageExtensionHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package magpiebridge.core;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class DefaultLanguageExtensionHandler implements LanguageExtensionHandler {
```
Comment (linghuiluo) [inline src/main/java/magpiebridge/core/DefaultLanguageExtensionHandler.java:12]: can you add class level java doc?

File: src/main/java/magpiebridge/core/LanguageExtensionHandler.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package magpiebridge.core;
+
+import java.util.Set;
+
+public interface LanguageExtensionHandler {
```
Comment (linghuiluo) [inline src/main/java/magpiebridge/core/LanguageExtensionHandler.java:10]: can you add class level java doc?

File: src/main/java/magpiebridge/file/SourceFileManager.java [hunk 4]
```diff
@@ -31,7 +36,7 @@
 public class SourceFileManager {
 
   /** The language. */
-  private String language;
+  public String language;
```
Comment (linghuiluo) [inline src/main/java/magpiebridge/file/SourceFileManager.java]: public field should be final

File: src/test/resources/DemoProjectNpm/package-lock.json [hunk 5]
```diff
@@ -1,8 +1,4500 @@
 {
```
Comment (linghuiluo) [inline src/test/resources/DemoProjectNpm/package-lock.json]: please don't change resource files

File: pom.xml (hunk 1 above)
Comment (julianthome) [reply pom.xml]: changed the version to v1.5.0 ✅

File: src/main/java/magpiebridge/core/DefaultLanguageExtensionHandler.java (hunk 2 above)
Comment (julianthome) [reply src/main/java/magpiebridge/core/DefaultLanguageExtensionHandler.java:12]: Added the class level Javadoc ✅

File: src/main/java/magpiebridge/file/SourceFileManager.java (hunk 4 above)
Comment (julianthome) [reply src/main/java/magpiebridge/file/SourceFileManager.java]: Ah thanks a lot for catching this -- this field was supposed to be private. Just changed that ✅

File: src/main/java/magpiebridge/core/LanguageExtensionHandler.java (hunk 3 above)
Comment (julianthome) [reply src/main/java/magpiebridge/core/LanguageExtensionHandler.java:10]: Javadoc comments added ✅

File: src/test/resources/DemoProjectNpm/package-lock.json (hunk 5 above)
Comment (julianthome) [reply src/test/resources/DemoProjectNpm/package-lock.json]: reverted back to the original file ✅

**Comment by julianthome** [comment]
Thank you very much @linghuiluo for the thorough review 👍🏼  I tried to address all your comments.

**Comment by linghuiluo** [comment]
@all-contributors add @julianthome for code and documentation

**Comment by linghuiluo** [comment]
@all-contributors please add @julianthome for code

**Pull request merged**